### PR TITLE
feat(design): add accesibility role to DaffListItemComponent

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
DaffListItemComponent is not assigned a role.

Fixes: N/A

## What is the new behavior?
Add `listitem` role for accessibility considerations.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/design/src/molecules/list/list-item/list-item.component.ts [hunk 1]
```diff
@@ -15,4 +15,22 @@ export class DaffListItemComponent {
...
   @ContentChild(DaffPrefixDirective, { static: false }) _prefix: DaffPrefixDirective;
   @ContentChild(DaffSuffixDirective, { static: false }) _suffix: DaffSuffixDirective;
+
+  constructor(private elementRef: ElementRef) {}
+
+  /**
+   * Sets the role for a regular `<daff-list-item>` to listitem.
+   */
+  @HostBinding('attr.role') get role() {
+    return this._isAnchor() ? null : 'listitem';
```
Comment (damienwebdev) [inline libs/design/src/molecules/list/list-item/list-item.component.ts:25]: Needs test coverage.

File: libs/design/src/molecules/list/list-item/list-item.component.ts [hunk 2]
```diff
@@ -15,4 +15,22 @@ export class DaffListItemComponent {
...
+  };
+
+  _getHostElement() {
+    return this.elementRef.nativeElement;
```
Comment (damienwebdev) [inline libs/design/src/molecules/list/list-item/list-item.component.ts:29]: private.

File: libs/design/src/molecules/list/list-item/list-item.component.ts (hunk 1 above)
Comment (xelaint) [reply libs/design/src/molecules/list/list-item/list-item.component.ts:25]: Updated.

File: libs/design/src/molecules/list/list-item/list-item.component.ts (hunk 2 above)
Comment (xelaint) [reply libs/design/src/molecules/list/list-item/list-item.component.ts:29]: Updated.

File: libs/design/src/molecules/list/list-item/list-item.component.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/design/src/molecules/list/list-item/list-item.component.ts:25]: What about the other side of the conditional?

Comment (xelaint) [reply libs/design/src/molecules/list/list-item/list-item.component.ts:25]: Updated.

**Pull request merged**